src/library/scala/collection/mutable/ArrayDeque.scala

```diff
@@ -421,7 +421,7 @@ class ArrayDeque[A] protected (
     res.result()
   }
 
-  override def reverse: IterableCC[A] = {
+  override def reverse: ArrayDeque[A] = {
```

Comment:
This one is problematic because subclasses of `ArrayDeque` will now only return an `ArrayDeque` instead of a more specific type.

Reply:
Don’t we normally declare concrete collections `final`?

Reply:
`ArrayDeque` is used to implement the mutable `Stack` and `Queue`.

Reply:
We can always make a new type alias within `ArrayDeque`, then it's more local. Or we create an `ArrayDequeOps` trait with a `CC` type param, `Stack` and `Queue` can then mix that in.